### ISSUE TITLE
Suggestion: Instead of triggering on join, could you make this an admin command?

### ISSUE THREAD

**Issue by Raclex** [SpitefulFox/Unthaumic#4]
I'm currently hosting a server and people are  on the fence on whether or not they want this in the pack. I want it for myself only. I would love a way to install this into the server and just run a command on myself instead of everyone getting it on join.

**Comment by SpitefulFox** [comment]
Can just spawn in the creative-mode thaumonomicon for yourself if you only want the effect for yourself. Look in the Thaumcraft config for the line about the cheater book.

**Comment by Raclex** [comment]
The main reason why I love your mod so much is because it then allows me to check the NEI for what objects have what aspects. Using the creative-mode thaumonomicon does not achieve this. I've also searched through Thaumcraft's built in commands and do not see a way to accomplish this without your mod. I've attempted to branch your mod and try to recreate it, but it appears you used Intellij, and I haven't done Java in a while, so was hoping you would be kind enough to bang out something really quickly. I'm sure you could.

**Comment by Raclex** [comment]
Nvm I think I have a working solution. Sorry to bother you.

**Comment by SpitefulFox** [comment]
Ah, I guess that's a pretty good point about the scanning thing.  Lemme know if you need it after all and maybe I'll come up with something for you. Maybe.  I've been pretty busy and haven't had much time for modding lately.

**Comment by Raclex** [comment]
I'm a programmer by trade and I was trying to work up a solution myself, but I couldn't even get the base mod to compile. I haven't done Java in a while and I've never messed with gradle before now, but I think if I just run the mod while I'm the only one in the game, and then remove it, I should get the desired results, assuming the effects of item.setAutoUnlock(); do not persist after the mod is removed (I was going to test this next). If you're curious, my gradle build had issues with dependencies. It first didn't like that Baubles didn't have a version, which looked to me like a placeholder for a variable in the gradle build command, so I just modified the file to have the proper string version. That seemed to work, but then it had issues trying to find the thaumcraft.common dependencies, etc. Which to me didn't even seem to be in the base Thaumcraft API, so I wasn't sure where to continue from there. I'm sure I'm missing something simple and completely obvious. I never installed forge so I'm not sure if I needed that for a gradle build or not. That's where I was headed next, but I've kind of given up. 1.7.10 is so old now it is difficult to find support. I mean, honestly, if you have experience writing custom commands you have all the code you need. Just need to slap it together real fast, it seems. I only need the stuff that runs on join, and only for certain whitelisted users, but again, once you start catering to every bob and joe you set a precedence, I get it.

**Comment by SpitefulFox** [comment]
Having the mod installed when you log in and then removing it afterwards should do the trick.

Most Thaumcraft add-ons just have a deobfuscated copy of Thaumcraft's jar in their /lib/ folder in order to compile because the API is notoriously incomplete.

**Comment by Raclex** [comment]
So, this isn't a big deal, but now I'm curious. I used your exact git repository that I thought containted the full deobfuscated Thaumcraft API. I also used your exact build.gradle file. What then am I missing? If it is too difficult to explain that's fine too. I'm just curious. I'm noticing yours doesn't have a lib folder, so I'm guessing you compile this differently than your git?

**Comment by SpitefulFox** [comment]
The repo doesn't include my lib folder because it just contains an entire copy of the Thaumcraft mod and redistributing other people's mods wholecloth is kinda frowned upon.

Basically, to get the repo working:

Install the 1.7.10 Forge dev in a folder.
Unzip the repo contents into the folder to replace the build.gradle and the src folder.
Make a lib folder and put inside of it a deobfuscated copy of the Thaumcraft jar. If a deobfuscated Thaumcraft.jar isn't available, you can make one using BON.
Put a deobfuscated copy of Baubles in the lib folder.

Basically, the mod makes references to Thaumcraft's core code instead of the API, so a deobfuscated copy of the mod itself is needed.
